### PR TITLE
[ticket=RW-6672][ticket=RW-5607][risk=no] Sidebar active state refactoring and UX touch ups (animations, save panels, etc)

### PR DESCRIPTION
**Pull request by ericsong**
## Functional changes
1. Sidebar animations when transitioning between panels
    - Before: https://user-images.githubusercontent.com/2770197/116337517-3cb87d00-a7a8-11eb-871e-9e1edbe5fb4a.mov
    - After: https://user-images.githubusercontent.com/2770197/116337541-4510b800-a7a8-11eb-8a5d-3c78dbe4074b.mov
2. Remember which icon was last active and load that panel instead of defaulting to the help info panel 
    - Before: https://user-images.githubusercontent.com/2770197/116337572-55289780-a7a8-11eb-838a-8bbecaae301e.mov
    - After: https://user-images.githubusercontent.com/2770197/116337587-5a85e200-a7a8-11eb-9551-d1bd71ba59fe.mov
3. Do not close participant review annotation sidebar panel when switching participants
    - Before: https://user-images.githubusercontent.com/2770197/116337622-6671a400-a7a8-11eb-9d94-7e73febda710.mov
    - After: https://user-images.githubusercontent.com/2770197/116337635-6d98b200-a7a8-11eb-8434-d4d1a2715253.mov
4. Show workspace storage sidebar icon on all pages
5. Nicer looking scroll bar for all sidebar panels
    - Before: https://user-images.githubusercontent.com/2770197/116341801-8f496780-a7af-11eb-8568-0007133448c9.mp4
    - After: https://user-images.githubusercontent.com/2770197/116341814-94a6b200-a7af-11eb-87d9-d0cd73338d6b.mp4

## Refactoring changes
1. helpContentKey -> pageKey
    - The helpContentKey property was becoming a bit difficult to work with since it was being used for several things like determining which icons to render, what analytics label to use, and what info to load in the help tips component. I was running into issues where changing the helpContentKey to fix a bug in one of those use cases would cause a bug in a different use case.
    - I renamed helpContentKey to pageKey and slightly changed its usages to reflect that semantic difference. It feels more natural to me that the router is emitting information about a page (a unique page key) rather than a key into a dictionary for a specific component (help-tips) that the vast majority of other components would have no interest in.
    - Each of the previous users of the helpContentKey now looks at the pageKey and is responsible for figuring out how that maps to their desired behavior. 
2. Explicitly declare show icon condition for each sidebar icon and use pageKey to drive this where
possible.
    - Criteria is the only place where this isn't possible because the cohort builder component only
has a single route/page/pageKey but requires the criteria icon to conditionally render depending on what the user is currently doing in the cohort builder component.
    - This wasn't an issue for the concept set component because I was able to supply a different pageKey for the parts of concept set builder that doesn't need the concept set icon.
3. Remove setSidebarState/sidebarOpen props and drive sidebar open/close state from the activeIcon store

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Review by freemabd — COMMENTED** [review]
I'm gonna defer to @dolbeew on this as the UI/sidebar expert

File: ui/src/app/components/help-sidebar.tsx [hunk 1]
```diff
@@ -43,7 +44,7 @@ import {
 import {withRuntimeStore} from 'app/utils/runtime-utils';
 import {
   CompoundRuntimeOpStore,
-  compoundRuntimeOpStore,
+  compoundRuntimeOpStore, routeDataStore,
```
Comment (dolbeew) [inline ui/src/app/components/help-sidebar.tsx:47]: Nit: move to new line

File: ui/src/app/components/help-sidebar.tsx [hunk 2]
```diff
@@ -672,18 +639,19 @@ export const HelpSidebar = fp.flow(
               </h3>,
             renderBody: () =>
               <HelpTips allowSearch={false}
```
Comment (dolbeew) [inline ui/src/app/components/help-sidebar.tsx:641]: Looks like this shows the same content as the 'help' icon just without the search bar. Why do we need both on every page?

Comment (ericsong) [reply ui/src/app/components/help-sidebar.tsx:642]: Ah, this should always be showing notebooks help info actually. Thanks for the catch. Displaying it on every page is a change Lou asked for as we were thinking about how to further develop the workspace storage tab into something more interact with the user's actual google bucket.

Comment (dolbeew) [reply ui/src/app/components/help-sidebar.tsx:642]: Ah, ok. Makes sense.

**Review by dolbeew — APPROVED** [review]

**Pull request merged**